modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/ScalaSttpClientCodegen.java

```diff
@@ -87,4 +191,300 @@ public CodegenOperation fromOperation(String path,
+        private static final String JSON4S = "json4s";
+        private static final String CIRCE = "circe";
+
+        public JsonLibraryProperty() {
+            super("jsonLibrary", "Json library to use. Possible values are: json4s and circe.", JSON4S);
+        }
+
+        @Override
+        public void updateAdditionalProperties(Map<String, Object> additionalProperties) {
+            String value = getValue(additionalProperties);
+            if (value.equals(CIRCE) || value.equals(JSON4S)) {
+                additionalProperties.put(CIRCE, value.equals(CIRCE));
+                additionalProperties.put(JSON4S, value.equals(JSON4S));
+            } else {
+                IllegalArgumentException exception =
+                        new IllegalArgumentException("Invalid json library: " + value + ". Must be " + CIRCE + " " +
+                                "or " + JSON4S);
+                throw exception;
+            }
+        }
+    }
+
+    public static class PackageProperty extends StringProperty {
+
+        public PackageProperty() {
+            super("mainPackage", "Top-level package name, which defines 'apiPackage', 'modelPackage', " +
+                    "'invokerPackage'", DEFAULT_PACKAGE_NAME);
+        }
+
+        @Override
+        public void updateAdditionalProperties(Map<String, Object> additionalProperties) {
+            String mainPackage = getValue(additionalProperties);
+            if (!additionalProperties.containsKey(CodegenConstants.API_PACKAGE)) {
+                String apiPackage = mainPackage + ".api";
+                additionalProperties.put(CodegenConstants.API_PACKAGE, apiPackage);
+            }
+            if (!additionalProperties.containsKey(CodegenConstants.MODEL_PACKAGE)) {
+                String modelPackage = mainPackage + ".model";
+                additionalProperties.put(CodegenConstants.MODEL_PACKAGE, modelPackage);
+            }
+            if (!additionalProperties.containsKey(CodegenConstants.INVOKER_PACKAGE)) {
+                String invokerPackage = mainPackage + ".core";
+                additionalProperties.put(CodegenConstants.INVOKER_PACKAGE, invokerPackage);
+            }
+        }
+
+        public String getApiPackage(Map<String, Object> additionalProperties) {
+            return additionalProperties.getOrDefault(CodegenConstants.API_PACKAGE, DEFAULT_PACKAGE_NAME + ".api").toString();
+        }
+
+        public String getModelPackage(Map<String, Object> additionalProperties) {
+            return additionalProperties.getOrDefault(CodegenConstants.MODEL_PACKAGE, DEFAULT_PACKAGE_NAME + ".model").toString();
+        }
+
+        public String getInvokerPackage(Map<String, Object> additionalProperties) {
+            return additionalProperties.getOrDefault(CodegenConstants.INVOKER_PACKAGE, DEFAULT_PACKAGE_NAME + ".core").toString();
+        }
+    }
+
+    private static abstract class CustomLambda implements Mustache.Lambda {
```

Comment:
Note that at we have well-tested lambdas in the `org.openapitools.codegen.templating.mustache` package which handle camel case, title case, uppercase, lowercase, etc. We should probably move these to use the tested lambdas at some point.

Reply:
I agree. I just copy-pasted it from scala-akka generator AFAIR. We should change it there as well.